File: Host Name & IP/gethostname.py
import socket
import sys

def get_host_by_ip(ip_address):
    try:
        # Perform reverse DNS lookup to get host information
        host_info = socket.gethostbyaddr(ip_address)
    except socket.herror:
        print("Error: No such host found")
        sys.exit(1)
    
    # The primary hostname
    print(f"Host name: {host_info[0]}")

    # All IP addresses associated with the host (only one in most cases)
    print("Host IP Address:")
    print(ip_address)
    
    # Other addresses (aliases)
    print("Other addresses:")
    for alias in host_info[2]:
        print(alias)
    
    # Any aliases (other hostnames) for the host
    print("Other host names or aliases:")
    for alias in host_info[1]:
        print(alias)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python gethostbyname.py ipaddress")
        sys.exit(1)
    
    ip_address = sys.argv[1]
    get_host_by_ip(ip_address)

File: Host Name & IP/gethostname.c
#include<stdio.h>
#include<netdb.h>
#include<stdlib.h>
#include<arpa/inet.h>

int main(int argc, char *argv[])
{
	int i;
	struct hostent* host;
	struct in_addr ipaddr;
	if(argc!=2)
	{
		printf("Usage: ./gethostbyname ipaddress\n");
		exit(1);
	}
	inet_aton(argv[1],&ipaddr);
	host=gethostbyaddr(&ipaddr,sizeof(ipaddr), AF_INET); 
	if(host==NULL)
	{
		printf("Error: No Such Host Found\n");
		exit(1);
	}
	printf("Host name: %s\n", host->h_name);
	printf("Host IP Address: %s\n", inet_ntoa(*((struct in_addr*) host->h_addr))); 
	struct in_addr ** addr_list; 
	printf("Other addresses :\n");
	addr_list= (struct in_addr **)host->h_addr_list;
	for(i=0;addr_list[i]!=NULL;i++)
		printf("%s\n",inet_ntoa(*addr_list[i]));
	printf("Other host names or aliases :\n");
	char **names;
	names= (char **)host->h_aliases;
	for(i=0;names[i]!=NULL;i++)
		printf("%s\n",names[i]);
	return 0;
}


